record_web/lib/record_web.dart

```diff
@@ -145,15 +159,34 @@ class RecordPluginWeb extends RecordPlatform {
     return devices;
   }
 
+  double getMaxAmplitude() {
+    assert(analyser != null);
+    analyser?.getFloatFrequencyData(amplitudeDataArray!);
+    var maxAmplitude = amplitudeDataArray!.reduce(max);
+    return maxAmplitude;
+  }
+
   void _onStart(html.MediaStream stream) {
     _mediaRecorder = html.MediaRecorder(stream);
     _mediaRecorder?.addEventListener('dataavailable', _onDataAvailable);
     _mediaRecorder?.addEventListener('stop', _onStop);
     _mediaRecorder?.start();
-
+    createAudioContext(stream);
     _updateState(RecordState.record);
   }
 
+  createAudioContext(html.MediaStream stream) {
```

Comment:
Create an internal audio context to get rid optionals and assign it finally to the class member

Reply:
didn't catch on this part, what is needed here ?

Reply:
You're mixing optional and not null assertions.

Reply:
ok got it